diff --git a/sal/rtl/math.cxx b/sal/rtl/math.cxx
index 78df29105..6290b992e 100644
--- a/sal/rtl/math.cxx
+++ b/sal/rtl/math.cxx
@@ -235,8 +235,8 @@ double stringToDouble(CharT const* pBegin, CharT const* pEnd, CharT cDecSeparato
             {
                 if (bufsize > 256)
                 {
-                    stringInHeap = std::make_unique_for_overwrite<char[]>(bufsize);
-                    mapInHeap = std::make_unique_for_overwrite<const CharT* []>(bufsize);
+                    stringInHeap.reset(new char[bufsize]);
+                    mapInHeap.reset(new const CharT*[bufsize]);
                     string = stringInHeap.get();
                     map = mapInHeap.get();
                 }
